File: src/main/java/com/lsd/diagram/SvgConverter.java
package com.lsd.diagram;

import net.sourceforge.plantuml.FileFormatOption;
import net.sourceforge.plantuml.SourceStringReader;

import java.io.ByteArrayOutputStream;
import java.io.IOException;

import static lsd.format.xml.XmlPrettyPrinter.indentXml;
import static net.sourceforge.plantuml.FileFormat.SVG;
import static org.apache.commons.lang3.StringUtils.countMatches;

public class SvgConverter {

    public String convert(String markup) {
        var result = new StringBuilder();
        var newmarkup = countMatches(markup, "@startuml");
        for (int i = 0; i < newmarkup; i++) {
            var svg = createSvg(markup, i);
            result.append(indentXml(svg).orElse(svg));
        }
        return result.toString();
    }

    private String createSvg(String plantUmlMarkup, int pageNumber) {
        try (var os = new ByteArrayOutputStream()) {
            var reader = new SourceStringReader(plantUmlMarkup);
            reader.outputImage(os, pageNumber, new FileFormatOption(SVG, false));
            return os.toString();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/com/lsd/diagram/SvgConverter.kt
package com.lsd.diagram

import lsd.format.xml.XmlPrettyPrinter
import net.sourceforge.plantuml.FileFormat
import net.sourceforge.plantuml.FileFormatOption
import net.sourceforge.plantuml.SourceStringReader
import org.apache.commons.lang3.StringUtils
import java.io.ByteArrayOutputStream

fun convertToSvg(markup: String): String {
    val result = StringBuilder()
    val umlCount = StringUtils.countMatches(markup, "@startuml")
    for (i in 0 until umlCount) {
        val svg = createSvg(markup, i)
        result.append(XmlPrettyPrinter.indentXml(svg).orElse(svg))
    }
    return result.toString()
}

private fun createSvg(plantUmlMarkup: String, pageNumber: Int): String {
    ByteArrayOutputStream().use { os ->
        val reader = SourceStringReader(plantUmlMarkup)
        reader.outputImage(os, pageNumber, FileFormatOption(FileFormat.SVG, false))
        return os.toString()
    }
}